### PR TITLE
Add secret utils python file

### PR DESCRIPTION
**Pull request by chunfuwen**
Pull up some distributed secret related functions into this file as common utils

Signed-off-by: chunfuwen <chwen@redhat.com>

### CODE REVIEW

**Comment by chunfuwen** [comment]
>>> from virttest import utils_secret
>>> utils_secret.clean_up_secrets()
>>> sec_list = utils_secret.get_secret_list()
>>> sec_list
[]
>>>

File: virttest/utils_secret.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    :param remote_virsh: remote virsh shell session.
+    :return secret list
+    """
+    logging.info("Get secret list ...")
+    if remote_virsh:
+        secret_list_result = remote_virsh.secret_list()
+    else:
+        secret_list_result = virsh.secret_list()
+    secret_list = secret_list_result.stdout_text.strip().splitlines()
```
Comment (sathnaga) [inline virttest/utils_secret.py:22]: it is good to wrap it in `try..` block and catch any obvious error outputs and return proper result incase failure condition.

File: virttest/utils_secret.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    # First two lines contain table header followed by entries
+    # for each secret, such as:
+    #
+    # UUID                                  Usage
+    # --------------------------------------------------------------------------------
+    # b4e8f6d3-100c-4e71-9f91-069f89742273  ceph client.libvirt secret
+    secret_list = secret_list[2:]
```
Comment (sathnaga) [inline virttest/utils_secret.py:29]: same as above, keep this inside `try..` too.

File: virttest/utils_secret.py (hunk 1 above)
Comment (chunfuwen) [reply virttest/utils_secret.py:27]: updated

File: virttest/utils_secret.py (hunk 2 above)
Comment (chunfuwen) [reply virttest/utils_secret.py:34]: updated

File: virttest/utils_secret.py (hunk 1 above)
Comment (chunfuwen) [reply virttest/utils_secret.py:27]: updated

File: virttest/utils_secret.py (hunk 2 above)
Comment (chunfuwen) [reply virttest/utils_secret.py:34]: updated

File: virttest/utils_secret.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    logging.info("Get secret list ...")
+    try:
+        if remote_virsh:
+            secret_list_result = remote_virsh.secret_list()
+        else:
+            secret_list_result = virsh.secret_list()
+    except Exception as e:
+        logging.warn("Exception thrown while getting secret lists: %s", str(e))
+
+    secret_list = secret_list_result.stdout_text.strip().splitlines()
```
Comment (kylazhang) [inline virttest/utils_secret.py:26]: If you can not get secret_list_result, then the following operation also will raise error, I think it is better to move them also in try.

File: virttest/utils_secret.py (hunk 2 above)
Comment (kylazhang) [reply virttest/utils_secret.py:33]: As above and give the secret_list a default value []

File: virttest/utils_secret.py (hunk 3 above)
Comment (chunfuwen) [reply virttest/utils_secret.py:26]: if error is thrown when secret_list = secret_list_result.stdout_text.strip().splitlines() , it is anticipated and the logic need break here.

File: virttest/utils_secret.py (hunk 2 above)
Comment (chunfuwen) [reply virttest/utils_secret.py:33]: secret_list  will become [] even if secret_list[2:] is out of index
>>> secret_list=[0,1]
>>> secret_list = secret_list[2:]
>>> secret_list
[]
>>>

**Comment by chunfuwen** [comment]
>>> from virttest import utils_secret
>>> utils_secret.get_secret_list
<function get_secret_list at 0x7feabe9ae400>
>>> bb=utils_secret.get_secret_list()
>>> bb
[]
>>> bb=utils_secret.clean_up_secrets()
>>> bb
>>> utils_secret.clean_up_secrets()
>>> bb=utils_secret.get_secret_list()
>>> bb
['a89d1730-ca47-468a-9dcb-aa6f46ac221c']
>>> utils_secret.clean_up_secrets()
>>> bb
['a89d1730-ca47-468a-9dcb-aa6f46ac221c']
>>> bb=utils_secret.get_secret_list()
>>> bb
[]
>>> cc=utils_secret.get_secret_list()
>>> c
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'c' is not defined
>>> cc
[]
>>> utils_secret.clean_up_secrets()
>>>

File: virttest/utils_secret.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    secret_list = get_secret_list(remote_virsh)
+    if secret_list:
+        for secret_uuid in secret_list:
+            try:
+                if remote_virsh:
+                    remote_virsh.secret_undefine(secret_uuid)
+                else:
+                    virsh.secret_undefine(secret_uuid)
+            except Exception as e:
+                logging.warn("Exception thrown while undefining secret: %s", str(e))
```
Comment (dzhengfy) [inline virttest/utils_secret.py:59]: Logging.error()?

**Comment by sathnaga** [comment]
@chunfuwen looks like the patchset needs a rebase against master( squash merge commits), 
warn can be replaced with error as suggest apart from that LGTM.

File: virttest/utils_secret.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+from virttest import virsh
+
+
+def get_secret_list(remote_virsh=None):
+    """
+    Get secret list by virsh secret-list from local or remote host.
+
+    :param remote_virsh: remote virsh shell session.
+    :return secret list.
```
[the remark this answers is not in the document]
Comment (chunfuwen) [reply virttest/utils_secret.py:15]: updated

File: virttest/utils_secret.py (hunk 4 above)
Comment (chunfuwen) [reply virttest/utils_secret.py:59]: updated

File: virttest/utils_secret.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    """
+    logging.info("Get secret list ...")
+    try:
+        if remote_virsh:
+            secret_list_result = remote_virsh.secret_list()
+        else:
+            secret_list_result = virsh.secret_list()
+    except Exception as e:
+        logging.warn("Exception thrown while getting secret lists: %s", str(e))
```
[the remark this answers is not in the document]
Comment (chunfuwen) [reply virttest/utils_secret.py:24]: updated

**Comment by chunfuwen** [comment]
miss repository name

**Pull request closed without merging**